### ISSUE TITLE
.env file variables undefined in build

### ISSUE THREAD

**Issue by tboulis** [electron-userland/electron-builder#4462]
* **Version**:  21.2.0

* **Target**: MacOS, Windows

Im using a `.env` file in the root of my project to store variables like API_URL etc, for different environments(dev, stg, prod). I use `dotenv` and initialize it at the start of my public/electron.js file. When I run the app locally for dev, env variables are correct. When i build the app they are undefined.   
   
I know I could use `.js` files for these variables, but I use `travis-ci` for building and testing my application, so I use its env (from settings) for different branches. I've searched a lot in the issues and on Stack Overflow but didn't find something similar. Is this a no-go practice with electron apps in general or am I doing something wrong?

**Comment by kzimny** [comment]
How do you compile your project for production build?
Maybe this description will help: https://github.com/motdotla/dotenv#how-do-i-use-dotenv-with-import

**Comment by tboulis** [comment]
@kzimny I just run ` electron-builder --publish=always`. Im using dotenv as described, I think the problem may be the fact that travis Env variables are available only at compilation of the build, so I will try and use a sed-like command for windows and sed for mac to add them in a .env file before running the build command on travis

**Comment by kzimny** [comment]
That should work.

**Comment by tboulis** [comment]
I did some tests and figured it out. I did use a script to add all environmental variables into a `.env` file which should be located inside my /public not in the root of the project (weirdly electron-builder didn't pick it up from the root). Here is the .travis.yml in case someone needs it:
```
os:
  - windows
  - osx
language: node_js
node_js: 12.13.0
cache:
  yarn: true
before_script:
  - cp public/.env.example public/.env
  - printenv > public/.env;
script:
  - |
    if [ "$TRAVIS_PULL_REQUEST" == "false" ] && ([ "$TRAVIS_BRANCH" == "master" ] || [ "$TRAVIS_BRANCH" == "development" ]);
    then yarn release;
    else yarn test;
    fi
```
A more elegant way to do this, is use the `.env.example` and get only its containing variables from the env of the vm, but I wasted a lot of time with `sed` and `awk` trying to do it, so I abandoned it. If someone does this it would be great.
